tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,360 @@
+
+inner = [['innerA'],
+         ['innerB']]
+outer = [['upleft',  inner],
+          ['lowleft', 'lowright']]
+
+fig, axd = plt.subplot_mosaic(outer, constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]')
+
+############################################################################
+# Low-level and advanced grid methods
+# ===================================
+#
+# `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` provide high-level
+# interfaces to create `~.gridspec.GridSpec` objects and add associated
+# new Axes to a figure at given `~.gridspec.SubplotSpec` inside the
+# *GridSpec*.
+#
+# Basic 2x2 grid
+# --------------
+#
+# We can accopmplish a 2x2 grid in the same manner as
+# ``plt.subplots(2, 2)``:
+
+fig = plt.figure(figsize=(4.5, 3.5), constrained_layout=True)
+spec = fig.add_gridspec(ncols=2, nrows=2)
+ax0 = fig.add_subplot(spec[0, 0])
+annotate_axes(ax0, 'ax0')
+ax1 = fig.add_subplot(spec[0, 1])
+annotate_axes(ax1, 'ax1')
+ax2 = fig.add_subplot(spec[1, 0])
+annotate_axes(ax2, 'ax2')
+ax3 = fig.add_subplot(spec[1, 1])
+annotate_axes(ax3, 'ax3')
+fig.suptitle('Manually added subplots using add_gridspec')
+
+#############################################################################
+# Axes spanning rows or grids in a grid
+# -------------------------------------
+#
+# We can index the *spec* array using `NumPy slice syntax
+# <https://numpy.org/doc/stable/reference/arrays.indexing.html>`_
+# and the new Axes will span the slice:
+
+fig = plt.figure(figsize=(4.5, 3.5), constrained_layout=True)
+spec = fig.add_gridspec(2, 2)
+ax0 = fig.add_subplot(spec[0, :])
+annotate_axes(ax0, 'ax0')
+ax10 = fig.add_subplot(spec[1, 0])
+annotate_axes(ax10, 'ax10')
+ax11 = fig.add_subplot(spec[1, 1])
+annotate_axes(ax11, 'ax11')
+fig.suptitle('Manually added subplots, spanning a column')
+
+###############################################################################
+# Manual Adjustments to a Gridspec Layout
+# ---------------------------------------
+#
+# When a GridSpec is explicitly used, you can adjust the layout
+# parameters of subplots that are created from the GridSpec.  Note this
+# option is not compatible with ``constrained_layout`` or
+# `.Figure.tight_layout` which both ignore *left* and *right* and adjust
+# subplot sizes to fill the figure.  Usually such manual placement
+# requires iterations to make the Axes tick labels not overlap the Axes.
+
+fig = plt.figure(constrained_layout=False, facecolor='0.9')
+gs = fig.add_gridspec(nrows=3, ncols=3, left=0.05, right=0.75,
+                      hspace=0.1, wspace=0.05)
+ax0 = fig.add_subplot(gs[:-1, :])
+annotate_axes(ax0, 'ax0')
+ax1 = fig.add_subplot(gs[-1, :-1])
+annotate_axes(ax1, 'ax1')
+ax2 = fig.add_subplot(gs[-1, -1])
+annotate_axes(ax2, 'ax2')
+fig.suptitle('Manual gridspec with right=0.75')
+
+###############################################################################
+# Nested layouts with SubplotSpec
+# ===============================
```

Comment:
```suggestion
# -------------------------------
```
I think this is still meant to be a subsection of "Low-level and advanced grid methods.